embassy-stm32/src/usb/mod.rs

```diff
@@ -26,13 +26,23 @@ fn common_init<T: Instance>() {
     // Clock might not be exact 48Mhz due to rounding errors in PLL calculation, or if the user
     // has tight clock restrictions due to something else (like audio).
     #[cfg(not(stm32h7rs))]
+    #[cfg(not(all(stm32u5, peri_usb_otg_hs)))]
     if freq.0.abs_diff(48_000_000) > 120_000 {
         panic!(
             "USB clock should be 48Mhz but is {} Hz. Please double-check your RCC settings.",
             freq.0
         )
     }
 
+    // For OTG-HS on STM32U5 only the 32MHz clock is fast enough (Table 762, Sect 73.14.4)
```

Comment:
I recall that in the general datasheet of the U5 it mentioned the HSE clock explicitly needed as well but that might be an implementation detail not relevant here but rather in RCC. Just wanted to write it down so I don't forget about it during testing.

Reply:
That's the reason I put the explicit reference. I thought this tied together with the otgphycr issue above, and the table only gives a value for 30MHz+ entry.  I may have wrapped this around my neck but the PLL1_P was also setting the AHB frequency?

Reply:
USB requires a crystal oscillator (so, HSE) because RC oscillators (like MSI, HSI) are not precise enough to meet the USB spec. This is the case for all chips, and both FS and HS.

(there are exceptions, many stm32 chips have some mechanism to calibrate HSI/MSI against either USB SOF or LSE which does make them precise enough)

Currently Embassy doesn't enforce this, on any chips. Not sure if we should, using USB from HSI does work in practice and if the user really wants to violate the spec we should let them (so it should have an opt-out at least). Also it's not trivial to implement. So IMO it's out of scope for this PR, and if we do it we should do it for all chips.

Reply:
OK, I think it's time for me to throw this one over the wall now for others with more experience to look at. For me it works. Outstanding issue just really seems to be to bring the PLL mapping over from the H7RS code but I quickly got myself in a mess with that.

Reply:
There are some enums and defines still missing in the pac for the U5 series that I can add later. Otherwise this looks pretty good imo.